src/python/pants/backend/jvm/tasks/scalafix.py

```diff
@@ -50,9 +50,31 @@ def source_extension(cls):
   @classmethod
   def prepare(cls, options, round_manager):
     super().prepare(options, round_manager)
-    # Only request a classpath if semantic checks are enabled.
+    # Only request a classpath and zinc_args if semantic checks are enabled.
     if options.semantic:
+      round_manager.require_data('zinc_args') 
       round_manager.require_data('runtime_classpath')
+ 
+  def execute(self): 
+    self.scalac_args = []
```

Comment:
Rather than using a mutable property pre-execute like this, please swap this to using `@memoized_property` (approximately Scala's `lazy val`) and then lazily computing the value:
```
@memoized_property
def _scalac_args(self):
  if self.get_options().semantic:
  ...
```

Reply:
Changed